client/network/src/service/out_events.rs

```diff
@@ -191,17 +179,16 @@ impl OutChannels {
 	/// Sends an event.
 	pub fn send(&mut self, event: Event) {
 		self.event_streams.retain_mut(|sender| {
-			let queue_size = sender.queue_size.fetch_add(1, Ordering::Relaxed);
-			if queue_size == sender.queue_size_warning && !sender.warning_fired {
+			if sender.inner.len() == sender.queue_size_warning && !sender.warning_fired {
 				sender.warning_fired = true;
 				sender.creation_backtrace.resolve();
 				error!(
-					"The number of unprocessed events in channel `{}` reached {}.\n\
+					"The number of unprocessed events in channel `{}` exceeded {}.\n\
 					 The channel was created at:\n{:?}",
```

Comment:
Hm.... shouldn't the backtrace be printed out with `{}` instead of `{:?}`?

Maybe also print out the backtrace from which this send is being made? I can imagine that could also be useful in some cases. It's a one time warning that should only trigger in exceptional cases, so I think we can afford it.

Reply:
> `backtrace::Backtrace` doesn't implement `std::fmt::Display`

I see. The one from `std` does though, if we'd switch to it.

Reply:
The `std` backtrace is resolved at the moment it's captured, so it was proposed by @bkchr to use `backtrace::Backtrace`, which can be resolved only if needed.

Reply:
> The `std` backtrace is resolved at the moment it's captured

Is it? I just checked before proposing it, and it doesn't seem to be? Unless I'm reading it wrong.

https://github.com/rust-lang/rust/blob/72067c77bdc1e8e339b9ed378a2c0ca0a9367c4d/library/std/src/backtrace.rs#L350

Here's where it's resolved:

https://github.com/rust-lang/rust/blob/72067c77bdc1e8e339b9ed378a2c0ca0a9367c4d/library/std/src/backtrace.rs#L443

And that's called only when `frames` are called (so only when the backtrace is actually accessed, which is from the `Display` impl below):

https://github.com/rust-lang/rust/blob/72067c77bdc1e8e339b9ed378a2c0ca0a9367c4d/library/std/src/backtrace.rs#L379

Am I reading this wrong? (:

Reply:
> Maybe also print out the backtrace from which this send is being made? I can imagine that could also be useful in some cases. It's a one time warning that should only trigger in exceptional cases, so I think we can afford it.

Done.

Reply:
> Is it? I just checked before proposing it, and it doesn't seem to be? Unless I'm reading it wrong.

Didn't check myself :see_no_evil:  Will fix in a moment :)

Reply:
Done